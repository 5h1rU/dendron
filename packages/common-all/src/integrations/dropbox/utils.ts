import { Note } from "../../node";
import { files } from "dropbox";
import path from "path";

export function denddronId2DxId(did: string): string {
  return `id:${did}`;
}
export function dxId2DendronId(dxId: string): string {
  return dxId.slice(3);
}

export function fileNameToTitle(name: string): string {
  const ext = path.extname(name);
  return path.basename(name, ext);
}

export function fileNameToTreePath(name: string): string[] {
  const title = fileNameToTitle(name);
  return title.split(".");
}

export function fileToNote(
  meta:
    | files.FileMetadata
    | files.FileMetadataReference
    | files.FolderMetadataReference,
  body?: string
) {
  const title = fileNameToTitle(meta.name);
  const id = dxId2DendronId(meta.id);
  const note = new Note({
    id,
    title,
    desc: "TODO",
    type: "note",
    schemaId: "-1",
    body
  });
  return note;
}
