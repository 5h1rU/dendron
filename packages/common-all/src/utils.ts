import minimatch from "minimatch";

export class DUtils {
  static minimatch = minimatch;
}
