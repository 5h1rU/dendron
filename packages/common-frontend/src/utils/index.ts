export * from "./logger";
export * from "./vscode";
