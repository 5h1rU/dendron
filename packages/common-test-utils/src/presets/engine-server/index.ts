import { ENGINE_DELETE_PRESETS } from "./delete";
import ENGINE_MULTI_TEST_PRESET from "./engine-multi";
import ENGINE_SINGLE_TEST_PRESET from "./engine-single";
import { ENGINE_GET_NOTE_BY_PATH_PRESETS } from "./getByPath";
import { ENGINE_INIT_PRESETS } from "./init";
import NOTE_REF from "./note-refs";
import { ENGINE_QUERY_PRESETS } from "./query";
import { ENGINE_RENAME_PRESETS } from "./rename";
import { ENGINE_UPDATE_PRESETS } from "./update";
import { ENGINE_WRITE_PRESETS } from "./write";

export const ENGINE_SERVER = {
  NOTE_REF,
  ENGINE_MULTI_TEST_PRESET,
  ENGINE_SINGLE_TEST_PRESET,
  ENGINE_WRITE_PRESETS,
  ENGINE_INIT_PRESETS,
  ENGINE_UPDATE_PRESETS,
  ENGINE_DELETE_PRESETS,
  ENGINE_GET_NOTE_BY_PATH_PRESETS,
  ENGINE_RENAME_PRESETS,
  ENGINE_QUERY_PRESETS,
};

export const ENGINE_PRESETS = [
  // {name: "init", presets: ENGINE_SERVER.ENGINE_INIT_PRESETS},
  // {name: "delete", presets: ENGINE_SERVER.ENGINE_DELETE_PRESETS},
  // {name: "getByPath", presets: ENGINE_SERVER.ENGINE_GET_NOTE_BY_PATH_PRESETS},
  { name: "query", presets: ENGINE_SERVER.ENGINE_QUERY_PRESETS },
  // {name: "rename", presets: ENGINE_SERVER.ENGINE_RENAME_PRESETS},
  // {name: "update", presets: ENGINE_SERVER.ENGINE_UPDATE_PRESETS},
  // {name: "write", presets: ENGINE_SERVER.ENGINE_WRITE_PRESETS},
];
