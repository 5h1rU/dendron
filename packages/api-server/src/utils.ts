import { DEngineV2 } from "@dendronhq/common-all/src";
import { DendronEngineV2 } from "@dendronhq/engine-server";
import _ from "lodash";
import { MemoryStore } from "./store/memoryStore";

export function getWSKey(uri: string) {
  return _.trimEnd(uri, "/");
}

export async function putWS({
  ws,
  engine,
}: {
  ws: string;
  engine: DendronEngineV2;
}) {
  MemoryStore.instance().put(`ws:${getWSKey(ws)}`, engine);
}

export async function getWS({ ws }: { ws: string }) {
  const engine = await MemoryStore.instance().get<DEngineV2>(
    `ws:${getWSKey(ws)}`
  );
  if (!engine) {
    throw `No Engine: ${ws}`;
  }
  return engine;
}
