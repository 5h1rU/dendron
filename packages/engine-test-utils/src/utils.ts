import { AssertUtils } from "@dendronhq/common-test-utils";
import { Git } from "@dendronhq/engine-server";
import fs from "fs-extra";
import path from "path";

export async function checkString(body: string, ...match: string[]) {
  expect(
    await AssertUtils.assertInString({
      body,
      match,
    })
  ).toBeTruthy();
}

export async function checkNotInString(body: string, ...nomatch: string[]) {
  expect(
    await AssertUtils.assertInString({
      body,
      nomatch,
    })
  ).toBeTruthy();
}

export class GitTestUtils {
  static async createRepoForWorkspace(wsRoot: string) {
    const git = new Git({ localUrl: wsRoot });
    await git.init();
    await git.add("dendron.yml");
    await git.commit({ msg: "init" });
  }
  static async createRepoWithReadme(root: string) {
    const git = new Git({ localUrl: root });
    await git.init();
    const readmePath = path.join(root, "README.md");
    fs.ensureFileSync(readmePath);
    await git.add(".");
    await git.commit({ msg: "init" });
  }
}
