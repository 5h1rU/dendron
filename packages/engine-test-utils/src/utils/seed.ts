import { DEngineClient, SeedCommands, SeedConfig } from "@dendronhq/common-all";
import { tmpDir, writeYAML } from "@dendronhq/common-server";
import { SeedCLICommand } from "@dendronhq/dendron-cli";
import { SeedInitMode, SeedUtils } from "@dendronhq/engine-server";
import path from "path";
import { GitTestUtils } from "./git";

export class TestSeedUtils {
  static async createSeedRegistry(opts: {
    engine: DEngineClient;
    wsRoot: string;
  }) {
    const config = await this.createSeed(opts);
    const id = SeedUtils.getSeedId(config);
    const root = tmpDir().name;
    const registryFile = path.join(root, "reg.yml");
    writeYAML(registryFile, { [id]: config });
    return { registryFile };
  }
  static async createSeed(opts: { engine: DEngineClient; wsRoot: string }) {
    const cli = new SeedCLICommand();
    const cmd = SeedCommands.INIT;
    const id = "dendron.foo";
    const seed: SeedConfig = {
      description: "",
      license: "",
      name: "foo",
      publisher: "dendron",
      repository: {
        type: "git",
        url: `file://${opts.wsRoot}`,
      },
      root: "vault",
    };
    await cli.execute({
      cmd,
      id,
      server: {} as any,
      config: seed,
      mode: SeedInitMode.CREATE_WORKSPACE,
      ...opts,
    });
    await GitTestUtils.addRepoToWorkspace(opts.wsRoot);
    return seed;
  }
}
