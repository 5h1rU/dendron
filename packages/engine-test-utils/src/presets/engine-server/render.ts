import { TestPresetEntryV4 } from "@dendronhq/common-test-utils";
import { ENGINE_HOOKS } from "./utils";

const NOTES = {
  BASIC: new TestPresetEntryV4(
    async ({ engine }) => {
      const { data } = await engine.renderNote({
        id: "foo",
        format: "markdown",
      });
      return [{ actual: data, expected: "<p>foo body</p>", msg: "foo" }];
    },
    {
      preSetupHook: async (opts) => {
        return ENGINE_HOOKS.setupBasic(opts);
      },
    }
  ),
};

export const ENGINE_RENDER_PRESETS = {
  NOTES,
};
