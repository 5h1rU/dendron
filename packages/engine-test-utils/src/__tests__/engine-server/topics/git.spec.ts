import { tmpDir } from "@dendronhq/common-server";
import { Git } from "@dendronhq/engine-server";
import { GitTestUtils } from "../../../utils";

describe("isRepo", async () => {
  test("no repo", async () => {
    const root = tmpDir().name;
    const repo = await Git.getRepo(root);
    expect(repo).toMatchSnapshot();
    expect(repo).toBeFalsy();
  });

  test("yes repo", async () => {
    const root = tmpDir().name;
    await GitTestUtils.createRepoWithReadme(root);
    const repo = await Git.getRepo(root);
    expect(repo).toMatchSnapshot();
    expect(repo).toBeTruthy();
  });
});
