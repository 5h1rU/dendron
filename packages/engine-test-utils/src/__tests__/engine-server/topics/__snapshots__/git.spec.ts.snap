// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isRepo no repo 1`] = `false`;

exports[`isRepo yes repo 1`] = `true`;
